internal/config/config.go

```diff
@@ -0,0 +1,218 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package config
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+
+	"sigs.k8s.io/kubebuilder/pkg/scaffold/resource"
+	"sigs.k8s.io/yaml"
+)
+
+// constants for scaffolding version
+const (
+	Version1 = "1"
+	Version2 = "2"
+
+	DefaultPath = "PROJECT"
+)
+
+// Config is the unmarshalled representation of the configuration file
+type Config struct {
+	// Path stores where the config was loaded from
+	path string
+
+	// Version is the project version, defaults to "1" (backwards compatibility)
+	Version string `json:"version,omitempty"`
+
+	// Domain is the domain associated with the project and used for API groups
+	Domain string `json:"domain,omitempty"`
+
+	// Repo is the go package name of the project root
+	Repo string `json:"repo,omitempty"`
+
+	// Resources tracks scaffolded resources in the project
+	// This info is tracked only in project with version 2
+	Resources []GVK `json:"resources,omitempty"`
+
+	// Multigroup tracks if the project has more than one group
+	MultiGroup bool `json:"multigroup,omitempty"`
+}
+
+func exists(path string) (bool, error) {
+	// Look up the file
+	_, err := os.Stat(path)
+
+	// If we could find it the file exists
+	if err == nil {
+		return true, nil
+	}
+
+	// Not existing and different errors are differentiated
+	if os.IsNotExist(err) {
+		err = nil
+	}
+	return false, err
+}
+
+// Exists verifies that the configuration file exists in the default path
+func Exists() (bool, error) {
+	return exists(DefaultPath)
+}
+
+// ExistsIn verifies that the configuration file exists in the provided path
+func ExistsIn(path string) (bool, error) {
+	return exists(path)
+}
+
+func load(path string) (*Config, error) {
+	// Read the file
+	in, err := ioutil.ReadFile(path) // nolint: gosec
+	if err != nil {
+		return &Config{}, err
+	}
+
+	// Unmarshal the file content
+	config := &Config{}
+	if err = yaml.Unmarshal(in, config); err != nil {
+		return &Config{}, err
+	}
+
+	// Store where it was loaded from
+	config.path = path
+
+	// kubebuilder v1 omitted version, so default to v1
+	if config.Version == "" {
+		config.Version = Version1
+	}
+
+	return config, nil
+}
+
+// Load obtains the configuration from the default path
+func Load() (*Config, error) {
```

Comment:
May:
```suggestion
func NewConfig() (*Config, error) {
```

Reply:
`NewConfig`? As I told you I'm trying to avoid the word "file"

Reply:
First, it should be `config.New` or `config.NewFromFile`:

> The importer of a package will use the name to refer to its contents, so exported names in the package can use that fact to avoid stutter. (Don't use the import . notation, which can simplify tests that must run outside the package they are testing, but should otherwise be avoided.) For instance, the buffered reader type in the bufio package is called Reader, not BufReader, because users see it as bufio.Reader, which is a clear, concise name. Moreover, because imported entities are always addressed with their package name, bufio.Reader does not conflict with io.Reader. Similarly, the function to make new instances of ring.Ring—which is the definition of a constructor in Go—would normally be called NewRing, but since Ring is the only type exported by the package, and since the package is called ring, it's called just New, which clients of the package see as ring.New. Use the package structure to help you choose good names.
[Source](https://golang.org/doc/effective_go.html#package-names)

Reply:
I was modifying this and didn't like it. `config.New` seems to be creating the configuration, but it has actually been created by a previous `kubebuilder init ...` command. It is creating a new struct from the configuration file, but the name is confusing, as the configuration itself is not being created.

For example, I would expect `kubebuilder init ...` to call `config.New` to create a new configuration file but I would not expect any other command after that to call `config.New`.